migrations/20250227_viz_move/migration.py

```diff
@@ -0,0 +1,249 @@
+    for folder_path in all_folders:
+        folder_name = os.path.basename(folder_path)
+        if folder_name == "network":
+            parent_folder = os.path.dirname(folder_path)
+
+            # Process CSV files
+            move_csv_files(folder_path, parent_folder, logger)
+
+            # Process GraphML files
+            move_graphml_files(folder_path, parent_folder, logger)
+
+
+def move_csv_files(network_folder, parent_folder, logger):
+    """Move CSV files from network folder to visualise folders."""
+    for file in os.listdir(network_folder):
+        if file.endswith(".csv"):
+            for dir_name in os.listdir(parent_folder):
+                if os.path.isdir(
+                    os.path.join(parent_folder, dir_name)
+                ) and dir_name.startswith("visualise_"):
+                    # Rename file
+                    new_fname = file.replace("network", dir_name)
+                    new_fname_path = os.path.join(network_folder, new_fname)
+
+                    if not os.path.exists(new_fname_path):
+                        shutil.copyfile(
+                            os.path.join(network_folder, file), new_fname_path
+                        )
+                        logger.info(f"Copied file: {file} → {new_fname}")
+
+                    # Move file to visualise folder
+                    visualise_folder = os.path.join(parent_folder, dir_name)
+                    if not os.path.exists(os.path.join(visualise_folder, new_fname)):
+                        shutil.move(new_fname_path, visualise_folder)
+                        logger.info(
+                            f"Moved file: {new_fname} → {visualise_folder}"
+                        )
+
+
+def move_graphml_files(network_folder, parent_folder, logger):
+    """Move GraphML files from network folder to visualise folders."""
+    for file in os.listdir(network_folder):
+        if file.endswith(".graphml"):
+            # Get lowest cluster number (assigned cluster)
+            cluster_nums = re.findall(r"\d+", file)
+            if not cluster_nums:
+                logger.warning(f"No cluster numbers found in {file}")
+                continue
+
+            cluster_num = min(map(int, cluster_nums))
+            visualise_dir = f"visualise_{cluster_num}"
+            visualise_path = os.path.join(parent_folder, visualise_dir)
+            
+            if os.path.exists(visualise_path):
+                new_fname = file.replace("network", visualise_dir)
+                new_fname_path = os.path.join(network_folder, new_fname)
+                
+                if not os.path.exists(new_fname_path):
+                    os.rename(
+                        os.path.join(network_folder, file), new_fname_path
+                    )
+                    logger.info(f"Renamed file: {file} → {new_fname}")
+
+                # Move file to visualise folder
+                if not os.path.exists(os.path.join(visualise_path, new_fname)):
```

Comment:
Same question here - why not just rename and move in one go?

Reply:
as above comment im stupid